ArcGISRuntimeSDKQt_QMLSamples/Maps/ShowDeviceLocationUsingIndoorPositioning/ShowDeviceLocationUsingIndoorPositioning.qml

```diff
@@ -0,0 +1,198 @@
+            Text {
+                id: positionSourceText
+                visible: false
+            }
+            Text {
+                id: transmitterCountText
+                visible: false
+            }
+            Text {
+                id: horizontalAccuracyText
+                visible: false
+            }
+        }
+    }
+
+    // An IPS positioning feature table is stored with an IPS-enabled map. Each row in the table contains an indoor positioning file.
+    // The IndoorsLocationDataSource will use the most recently created row unless given an alternative GlobalId in the constructor.
+    function findPositioningTable(tableName) {
+        map.tables.forEach((table) => {
+                               table.loadStatusChanged.connect(() => {
+                                                                   if (table.loadStatus === Enums.LoadStatusLoaded && table.tableName === tableName) {
+                                                                       positioningTable = table;
+                                                                   }
+                                                               });
+                               table.load();
+                           });
+    }
+
+    // The pathways table is an ArcGISFeatureTable with line features that represent paths through the indoor space.
+    // Locations provided by the IndoorsLocationDataSource are snapped to the lines in this feature class.
+    function findPathwaysTable(layerName) {
+        for (let i = 0; i < map.operationalLayers.count; ++i) {
+            const layer = map.operationalLayers.get(i);
+            if (layer.name === layerName && layer.layerType === Enums.LayerTypeFeatureLayer) {
+                const table = layer.featureTable;
+                if (table.featureTableType === Enums.FeatureTableTypeServiceFeatureTable) {
+                    pathwaysTable = table;
+                    return;
+                }
+            }
+        }
+    }
+
+    function setupIndoors() {
+        if (!positioningTable || !pathwaysTable)
+            return;
+
+        const indoorsLDS = ArcGISRuntimeEnvironment.createObject("IndoorsLocationDataSource",
+                                                                 {
+                                                                     positioningTable: positioningTable,
+                                                                     pathwaysTable: pathwaysTable
+                                                                 });
+
+        indoorsLDS.errorChanged.connect(()=>
+                                        {
+                                            console.log("IndoorsLocationDataSource errorChanged", indoorsLDS.error.message, indoorsLDS.error.additionalMessage);
+                                        });
+
+        // Update information text box and floor view if necessary
+        mapView.locationDisplay.locationChanged.connect(() => {
```

Comment:
I'm just realizing this is somewhat redundant, but the data source itself (the `indoorsLDS`) also has a `locationChanged` event you can monitor.  I assume you meant this one, but it's worth pointing out.